rpcs3/Emu/Cell/lv2/sys_fs.cpp

```diff
@@ -2932,67 +3009,79 @@ error_code sys_fs_truncate2(ppu_thread&, u32 fd, u64 size)
 
 error_code sys_fs_get_mount_info_size(ppu_thread&, vm::ptr<u64> len)
 {
-	sys_fs.todo("sys_fs_get_mount_info_size(len=*0x%x)", len);
+	sys_fs.trace("sys_fs_get_mount_info_size(len=*0x%x)", len);
 
 	if (!len)
 	{
 		return CELL_EFAULT;
 	}
 
-	*len = 0x9;
+	*len = lv2_fs_object::get_mount_count();
 
 	return CELL_OK;
 }
 
-error_code sys_fs_get_mount_info(ppu_thread&, vm::ptr<CellFsMountInfo> info, u32 len, vm::ptr<u64> out_len)
+error_code sys_fs_get_mount_info(ppu_thread&, vm::ptr<CellFsMountInfo> info, u64 len, vm::ptr<u64> out_len)
 {
-	sys_fs.todo("sys_fs_get_mount_info(info=*0x%x, len=0x%x, out_len=*0x%x)", info, len, out_len);
+	sys_fs.trace("sys_fs_get_mount_info(info=*0x%x, len=0x%x, out_len=*0x%x)", info, len, out_len);
 
-	if (!out_len)
+	if (!info || !out_len)
 	{
 		return CELL_EFAULT;
 	}
 
-	// TODO there is a case where 'something' happens if !info or len == 0
-	if (!info || len == 0)
-	{
-		sys_fs.todo("sys_fs_get_mount_info special case TODO");
-	}
-
-	const u32 max_len = std::min<u32>(len, 9);
-	*out_len = max_len;
-
 	struct mount_info
 	{
-		std::string_view path, filesystem, dev_name;
+		std::string path, filesystem, dev_name;
```

Comment:
This change is not needed anymore actually. The lifetime of the created string ends only after push_info returns.

Reply:
Pretty right. Thanks for notifying!